src/Microsoft.ML.StaticPipe/TreeTrainersStatic.cs

```diff
@@ -45,16 +44,56 @@ public static class TreeRegressionExtensions
             int numTrees = Defaults.NumTrees,
             int minDatapointsInLeaves = Defaults.MinDocumentsInLeaves,
             double learningRate = Defaults.LearningRates,
-            Action<FastTreeRegressionTrainer.Arguments> advancedSettings = null,
             Action<FastTreeRegressionModelParameters> onFit = null)
         {
-            CheckUserValues(label, features, weights, numLeaves, numTrees, minDatapointsInLeaves, learningRate, advancedSettings, onFit);
+            CheckUserValues(label, features, weights, numLeaves, numTrees, minDatapointsInLeaves, learningRate, onFit);
 
             var rec = new TrainerEstimatorReconciler.Regression(
                (env, labelName, featuresName, weightsName) =>
                {
                    var trainer = new FastTreeRegressionTrainer(env, labelName, featuresName, weightsName, numLeaves,
-                       numTrees, minDatapointsInLeaves, learningRate, advancedSettings);
+                       numTrees, minDatapointsInLeaves, learningRate);
+                   if (onFit != null)
+                       return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                   return trainer;
+               }, label, features, weights);
+
+            return rec.Score;
+        }
+
+        /// <summary>
+        /// FastTree <see cref="RegressionContext"/> extension method.
+        /// Predicts a target using a decision tree regression model trained with the <see cref="FastTreeRegressionTrainer"/>.
+        /// </summary>
+        /// <param name="ctx">The <see cref="RegressionContext"/>.</param>
+        /// <param name="label">The label column.</param>
+        /// <param name="features">The features column.</param>
+        /// <param name="weights">The optional weights column.</param>
+        /// <param name="advancedSettings">Algorithm advanced settings.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained. Note that this action cannot change the result in any way;
+        /// it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The Score output column indicating the predicted value.</returns>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[FastTree](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Static/FastTreeRegression.cs)]
+        /// ]]></format>
+        /// </example>
+        public static Scalar<float> FastTree(this RegressionContext.RegressionTrainers ctx,
+            Scalar<float> label, Vector<float> features, Scalar<float> weights,
+            FastTreeRegressionTrainer.Options advancedSettings,
```

Comment:
Is it possible to remove these fields?

Reply:
@Tom.  

In the API which accepts the `Options` argument,  how do we handle the  arguments  `label`, `features`, `weights`  in the Static API ?

In the corresponding Dynamic API,  we were able to remove  the parameters  for `labelColumn`, `featureColumn` and `weightColumn`  because  these were already part of the `Options` class  


---
In reply to: [245972452](https://github.com/dotnet/machinelearning/pull/2047#discussion_r245972452) [](ancestors = 245972452)

Reply:
You obviously cannot remove the parameters for these *here*. The options object just accepts a `string`. The entire point of the statically typed API is that someone can define a pipeline in a way that is type safe. So, no, you are not removing these parameters.

---
In reply to: [246165575](https://github.com/dotnet/machinelearning/pull/2047#discussion_r246165575) [](ancestors = 246165575,245972452)

Reply:
Thanks Tom for the clarification.

---
In reply to: [246170641](https://github.com/dotnet/machinelearning/pull/2047#discussion_r246170641) [](ancestors = 246170641,246165575,245972452)